contracts/wrapper/Wrapper.sol

```diff
@@ -0,0 +1,85 @@
+//SPDX-License-Identifier: Apache-2.0
+pragma solidity 0.8.17;
+
+import "@equilibria/root/token/types/Token18.sol";
+import "@equilibria/root/token/types/Token6.sol";
+import "@equilibria/root/control/unstructured/UReentrancyGuard.sol";
+import "@equilibria/root/control/unstructured/UOwnable.sol";
+import "../interfaces/ITwoWayBatcher.sol";
+import "../interfaces/IWrapper.sol";
+
+/// @title Wrapper
+/// @notice Helper contract for wrapping and unwrapping USDC and DSU without worrying about approvals or dependences.
+/// @dev To wrap USDC, the caller must first send USDC to this contract, then call `wrap`.
+///      Similarly, to unwrap DSU, the caller must first send DSU to this contract, then call
+///      `unwrap`. This contract can be optionally supplied with a TwoWayBatcher to save gas. It is
+///      recommended to only use this contract in contexts where the caller can atomically bundle
+///      the USDC/DSU transfer with the `wrap`/`unwrap` call (e.g. if the caller is a contract that
+///      can bundle the transfer with the call in a single transaction, or if the caller is an EOA
+///      using flashbots-style bundles).
+contract Wrapper is IWrapper, UReentrancyGuard, UOwnable {
+    /// @dev Reserve address
+    IEmptySetReserve public immutable RESERVE; // solhint-disable-line var-name-mixedcase
+
+    /// @dev DSU address
+    Token18 public immutable DSU; // solhint-disable-line var-name-mixedcase
+
+    /// @dev USDC address
+    Token6 public immutable USDC; // solhint-disable-line var-name-mixedcase
+
+    /// @dev Batcher address
+    ITwoWayBatcher public batcher;
```

Comment:
I'm thinking maybe we should just make this immutable as well since we'll have to change the batcher-related code for different batcher implementations, so it will require a redeploy either way.

Maybe we should add a better interface to the batchers so we don't need to do that in the future 😓 .

Reply:
What if we deploy a wrapper before a batcher?